File: front-end/tour_and_travel/lib/models/chat_message.dart
class ChatMessage {
  final int id;
  final int? ticketId;
  final int senderId;
  final String message;
  final String? imageUrl;
  final DateTime timestamp;
  final bool isAdminMessage;
  final int? adminId;
  final String? adminName;
  final bool isRead;

  ChatMessage({
    required this.id,
    this.ticketId,
    required this.senderId,
    required this.message,
    this.imageUrl,
    required this.timestamp,
    required this.isAdminMessage,
    this.adminId,
    this.adminName,
    required this.isRead,
  });

  factory ChatMessage.fromJson(Map<String, dynamic> json) {
    return ChatMessage(
      id: json['id'] ?? 0,
      ticketId: json['ticketId'],
      senderId: json['senderId'] ?? 0,
      message: json['message'] ?? '',
      imageUrl: json['imageUrl'],
      timestamp: json['timestamp'] != null 
          ? DateTime.parse(json['timestamp']) 
          : DateTime.now(),
      isAdminMessage: json['isAdminMessage'] ?? false,
      adminId: json['adminId'],
      adminName: json['adminName'],
      isRead: json['isRead'] ?? false,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'ticketId': ticketId,
      'senderId': senderId,
      'message': message,
      'imageUrl': imageUrl,
      'timestamp': timestamp.toIso8601String(),
      'isAdminMessage': isAdminMessage,
      'adminId': adminId,
      'adminName': adminName,
      'isRead': isRead,
    };
  }
}

File: backend/Tour_&_Travel_api/Tour_&_Travel_api/Models/ChatMessage.cs
using System;

namespace TravelApp.Models
{
    public class ChatMessage
    {
        public int Id { get; set; }
        
        // Nullable TicketId to avoid crash on existing data, but we'll use it going forward
        public int? TicketId { get; set; }
        
        public int SenderId { get; set; } // User ID or Admin ID
        
        // ReceiverId is mostly legacy now since we use TicketId, but keep it
        public int ReceiverId { get; set; } 
        
        public string Message { get; set; } = string.Empty;
        public string? ImageUrl { get; set; } // Optional image attached
        
        public DateTime Timestamp { get; set; } = DateTime.UtcNow;
        
        public bool IsAdminMessage { get; set; } // True if sent by admin
        public int? AdminId { get; set; } // The actual admin who replied
        
        public bool IsRead { get; set; } = false;
        
        [System.Text.Json.Serialization.JsonIgnore]
        public SupportTicket? Ticket { get; set; }
    }
}
